### PR TITLE
[flux] Update to version 0.4.0

### LINKED ISSUE #42055: [flux] update to v0.4.0
### Library name

Flux

### New version number

0.4.0

### Other information that may be useful (release notes, etc...)

Release notes: https://github.com/tcbrindle/flux/releases/tag/v0.4.0

### PR DESCRIPTION
**Pull request by SidneyCogdill**
https://github.com/tcbrindle/flux/issues/206
Fixes https://github.com/microsoft/vcpkg/issues/42055
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/flux/vcpkg.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
   "name": "flux",
-  "version-date": "2023-08-25",
+  "version": "0.4.0",
   "description": "Experimental C++20 library that provides a broadly equivalent feature set to C++20 Ranges, but uses a slightly different model based around cursors rather than iterators. Safer, easier, more efficient, compatible.",
   "homepage": "https://github.com/tcbrindle/flux",
+  "builtin-baseline": "146dc5495a2e0add6d3cd34cb0b53277a8e5144c",
```
Comment (dg0yt) [inline ports/flux/vcpkg.json:6]: This has no effect in a port manifest.

Comment (SidneyCogdill) [reply ports/flux/vcpkg.json]: Removed now. It was automatically added by one of the vcpkg commands.

**Review by WangWeiLin-MV — DISMISSED** [review]
The port usage tests pass with the following triplets:

* x64-windows

**Comment by WangWeiLin-MV** [comment]
> * The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.

Please check the issue.

**Comment by SidneyCogdill** [comment]
Removing `builtin-baseline` with the web editor without re-running `vcpkg x-add-version` locally causes problems. The updated hash is now committed.

**Review by WangWeiLin-MV — APPROVED** [review]

**Pull request merged**